experimental/dds/tree2/src/test/feature-libraries/editable-tree-2/lazyField.spec.ts

```diff
@@ -65,3 +92,316 @@ describe("lazyField", () => {
 		);
 	});
 });
+
+function createSingleValueTree<Kind extends FieldKind, Types extends AllowedTypes>(
+	builder: SchemaBuilder,
+	rootSchema: FieldSchema<Kind, Types>,
+	initialTree?:
+		| SchemaAware.TypedField<FieldSchema, SchemaAware.ApiMode.Flexible>
+		| readonly ITreeCursorSynchronous[]
+		| ITreeCursorSynchronous,
+): {
+	context: Context;
+	cursor: ITreeSubscriptionCursor;
+	forest: IEditableForest;
+} {
+	const schema = builder.intoDocumentSchema(rootSchema);
+	const forest = forestWithContent({ schema, initialTree });
+
+	const context = getReadonlyContext(forest, schema);
+	const cursor = context.forest.allocateCursor();
+
+	assert.equal(
+		context.forest.tryMoveCursorToField(rootFieldAnchor, cursor),
+		TreeNavigationResult.Ok,
+	);
+
+	return {
+		forest,
+		context,
+		cursor,
+	};
+}
+
+describe("LazyOptionalField", () => {
+	describe("as", () => {
+		it("Any", () => {
+			const builder = new SchemaBuilder("test");
+			const booleanLeafSchema = builder.leaf("bool", ValueSchema.Boolean);
+
+			const { context, cursor } = createSingleValueTree(
+				builder,
+				SchemaBuilder.fieldOptional(builder.struct("struct", {})),
+				{},
+			);
+
+			const field = new LazyOptionalField(
+				context,
+				SchemaBuilder.fieldOptional(Any),
+				cursor,
+				detachedFieldAnchor,
+			);
+
+			// Positive cases
+			assert(field.is(SchemaBuilder.fieldOptional(Any)));
+			assert(field.is(SchemaBuilder.fieldRecursive(FieldKinds.optional, Any)));
+
+			// Negative cases
+			assert(!field.is(SchemaBuilder.fieldOptional()));
+			assert(!field.is(SchemaBuilder.fieldOptional(booleanLeafSchema)));
+			assert(!field.is(SchemaBuilder.fieldValue(Any)));
+			assert(!field.is(SchemaBuilder.fieldSequence(Any)));
+			assert(!field.is(SchemaBuilder.fieldRecursive(FieldKinds.optional, booleanLeafSchema)));
```

Comment:
this is the same as 
```suggestion
			assert(!field.is(SchemaBuilder.field(FieldKinds.optional, booleanLeafSchema)));
```

which is also the same as:

```suggestion
			assert(!field.is(SchemaBuilder.fieldOptional(booleanLeafSchema)));
```

so no reason to test them all.